### ISSUE TITLE
No state found for id 'finalized'

### ISSUE THREAD

**Issue by dapplion** [ChainSafe/lodestar#4254]
**Describe the bug**

- From a host running a metrics exporter https://github.com/samcm/ethereum-metrics-exporter in https://github.com/gnosischain/consensus-deployment-ansible/issues/9

```
Jun-30 12:29:19.698[API]             error: Req req-3fw getStateFinalityCheckpoints error  No state found for id 'finalized'
Error: No state found for id 'finalized'
    at resolveStateId (file:///usr/app/node_modules/@chainsafe/lodestar/src/api/impl/beacon/state/utils.ts:41:11)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at getState (file:///usr/app/node_modules/@chainsafe/lodestar/src/api/impl/beacon/state/index.ts:24:12)
    at Object.getStateFinalityCheckpoints (file:///usr/app/node_modules/@chainsafe/lodestar/src/api/impl/beacon/state/index.ts:39:21)
    at Object.handler (file:///usr/app/node_modules/@chainsafe/lodestar-api/src/utils/server/genericJsonServer.ts:39:23)
```

**Comment by philknows** [comment]
Duplicate of #4253

**Comment by dapplion** [comment]
This case is covered 

https://github.com/ChainSafe/lodestar/blob/83b29d7d8d15df844b450c9ca2850ceaea29c1d7/packages/beacon-node/src/api/impl/beacon/state/utils.ts#L140-L141

A fallback would be to load the state from disk if not found

**Comment by philknows** [comment]
Is this still an issue? I'm not seeing this anymore as of v1.11.3

**Comment by nflaig** [comment]
This is still a problem until a solution like https://github.com/ChainSafe/lodestar/pull/6033 is implemented, related issue https://github.com/ChainSafe/lodestar/issues/5846

```sh
curl -s http://localhost:9596/eth/v1/beacon/states/finalized/finality_checkpoints | jq
{
  "statusCode": 404,
  "error": "Not Found",
  "message": "No state found for id 'finalized'"
}
```

**Comment by wemeetagain** [comment]
closed since #6033 is merged

**Comment by nflaig** [comment]
just noting here that while the exact query of this issue passing 'finalized' returns a result now, we still can't query historical data past that as we don't use `getStateResponseWithRegen` consistently for all relevant apis


https://github.com/ChainSafe/lodestar/blob/0d87c28aedb2e0556b027fc0714349542ff17542/packages/beacon-node/src/api/impl/beacon/state/index.ts#L64
